### PR TITLE
Rewrite docker logic as ENV contract, fix bugs in task.service

### PR DESCRIPTION
**Pull request by vvkin**
1. Rewrite `docker.service` according to ENV contract
2. Improve metrics and logs collection in `docker`
3. Fix some task-related bugs
4. Replace integration tests for `task` with e2e tests

### CODE REVIEW

File: src/plugins/task/task.service.ts [hunk 1]
```diff
@@ -24,49 +21,73 @@ import {
...
     }
-    // await this.metricsService.save(rawStats, logs)
+
+    return runId;
+  }
+
+  registerEnvHandleListeners(id: TaskRunId, envHandle: EnvHandle): void {
+    const logGenerator = envHandle.logs();
+    const metricGenerator = envHandle.metrics();
+    setImmediate(async () => {
```
Comment (nosovk) [inline src/plugins/task/task.service.ts:70]: зачем тут setImmediate?

Comment (vvkin) [reply src/plugins/task/task.service.ts:70]: To run it in the background and start it as soon as possible. I wanted to start listening to two generators simultaneously

File: src/lib/random.utils.ts [hunk 2]
```diff
@@ -1,4 +1,11 @@
 import { randomUUID } from 'crypto';
```
Comment (nosovk) [inline src/lib/random.utils.ts]: тут тоже нужно node:crypto наверное

File: src/lib/random.utils.ts [hunk 3]
```diff
@@ -1,4 +1,11 @@
...
+import { randomInt } from 'node:crypto';
 
 export const generateRandomString = (prefix?: string): string =>
   (prefix ? `${prefix}_` : '') + randomUUID();
+
+export const generateRandomIp = (): string =>
+  Array.from({ length: 4 }, () => randomInt(0, 255)).join('.');
+
+export const generateRandomEmail = (): string =>
+  `${generateRandomString()}@prescott.dev`;
```
Comment (nosovk) [inline src/lib/random.utils.ts]: домен .dev существует, есть специальный домен .test лучше всегда его использовать

**Pull request merged**